packages/runtime/runtime-definitions/src/dataStoreRegistry.ts

```diff
@@ -20,11 +20,14 @@ export type FluidDataStoreRegistryEntry =
  * An associated pair of an identifier and registry entry.  Registry entries
  * may be dynamically loaded.
  */
-export type NamedFluidDataStoreRegistryEntry = [string, Promise<FluidDataStoreRegistryEntry>];
+export type NamedFluidDataStoreRegistryEntry =
+    [string, Promise<FluidDataStoreRegistryEntry> | FluidDataStoreRegistryEntry];
 /**
  * An iterable identifier/registry entry pair list
  */
-export type NamedFluidDataStoreRegistryEntries = Iterable<NamedFluidDataStoreRegistryEntry>;
+export type FluidDataStoreRegistryEntries = Iterable<NamedFluidDataStoreRegistryEntry | IProvideFluidDataStoreFactory>;
```

Comment:
i guess i'm failing to see why this is better, isn't there the same number of types as before?

Reply:
This looks more into the future. We can remove one more concept - PureDataObjectFactory.registryEntry, and instead just use factory.